src/main/java/com/capitalone/dashboard/controller/SonarController.java

```diff
@@ -75,35 +71,30 @@ public SonarController(SonarProjectRepository sonarProjectRepository,
         return sendResponse("sonar instance url is invalid");
     }
 
-    private void gatherParams(String projectName, String projectKey, String instanceUrl) {
-        this.projectName = projectName;
-        this.projectKey = projectKey;
-        this.instanceUrl = instanceUrl;
-    }
-
     private ResponseEntity<String> sendResponse(String message) {
         HttpHeaders httpHeaders = new HttpHeaders();
         httpHeaders.setContentType(MediaType.APPLICATION_JSON);
         return new ResponseEntity<>(message, httpHeaders, HttpStatus.OK);
     }
 
-    private SonarProject createNewProjectIfNotExists() {
-        SonarProject project = null;
-        if (Objects.nonNull(this.projectKey)) {
-            SonarProject newProject = this.sonarClient.getProject(this.projectKey, this.instanceUrl);
-            if (Objects.nonNull(newProject)) {
-                SonarProject existingProject = sonarProjectRepository.findSonarProject(this.collector.getId(),
-                        newProject.getInstanceUrl(), newProject.getProjectId());
-                project = Objects.nonNull(existingProject) ? existingProject : newProject;
+    private SonarProject getExistingProject(ObjectId collectorId, String instanceUrl, String projectName,
+                                            String projectKey, SonarClient sonarClient) {
+        if (Objects.nonNull(projectKey)) {
+            SonarProject project = sonarClient.getProject(projectKey, instanceUrl);
+            if (Objects.nonNull(project)) {
+                List<SonarProject> sonarProjects = sonarProjectRepository.
+                        findSonarProjectsByProjectName(collectorId, instanceUrl, projectName);
+                if (CollectionUtils.isNotEmpty(sonarProjects)) {
+                    sonarProjects = sonarProjects.stream().filter(p -> p.getProjectId()==null || project.getProjectId().equals(p.getProjectId())).collect(Collectors.toList());
+                }
+                if (CollectionUtils.isNotEmpty(sonarProjects)) {
+                    Collections.sort(sonarProjects, Comparator.comparing(SonarProject::getLastUpdated).reversed());
+                    SonarProject existingProject = sonarProjects.get(0);
+                    existingProject.setProjectId(project.getProjectId());
+                    return existingProject;
+                }
```

Comment:
Any reason for the duplicate if cases?

Reply:
Not a duplicate. The variable sonarProjects get further filtered (reassigned).

Reply:
Ahhh makes sense. Approved.